lib/Charts/ChartRenderer.js

```diff
@@ -0,0 +1,423 @@
+'use strict';
+
+import {mouse as d3Mouse, select as d3Select} from 'd3-selection';
+import {axisBottom as d3AxisBottom, axisLeft as d3AxisLeft} from 'd3-axis';
+import {transition as d3Transition} from 'd3-transition';
+
+import defaultValue from 'terriajs-cesium/Source/Core/defaultValue';
+import defined from 'terriajs-cesium/Source/Core/defined';
+
+import getUniqueValues from '../Core/getUniqueValues';
+import Scales from './Scales';
+import Size from './Size';
+import Title from './Title';
+import Tooltip from './Tooltip';
+import ChartType from './ChartType';
+
+import d3Sync from './d3Sync';
+
+//const yAxisLabelWidth = 21;
+
+const defaultMargin = {
+    top: 20,
+    right: 30,
+    bottom: 20,
+    left: 0
+};
+
+const yAxisLabelWidth = 21;
+
+const defaultNoDataText = 'No preview available';
+
+const defaultTransitionDuration = 1000; // milliseconds
+
+const threshold = 1e-8; // Threshold for 'equal' x-values during mouseover.
+
+class ChartRenderer {
```

Comment:
I haven't used `class` in js yet. According to the [docs](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes) don't you need a `constructor` for `new` to work?